### PR TITLE
Refactor Bech32 and TLV usage

### PR DESCRIPTION
**Pull request by joshuatbrown**
## Issues covered
#1294 

## Description
This removes direct usage of Bech32 and TLV in favor of NostrIdentifier. It removes TLV entirely, as it's been superseded by TLVElement. Now, the only places we use TLVElement are in NostrIdentifier. And the only place we use Bech32.decode is in NostrIdentifier.

There's still a Bech32.encode method that's used in three places. We could replace those with a not-yet-written encode method on NostrIdentifier. I think that would be fairly simple.

## How to test
1. Examine the unit tests. Run them, too.
2. Make sure the app still works, especially around keys and `npub`s and `note`s.

## To Do

- [x] Fix issue with `nostr:naddr` links jumping out to another app.
- [ ] Consider moving Bech32.encode into NostrIdentifier, and do it if it makes sense.
- [x] Rename NostrConstants.swift, or delete it and move the constants to NostrIdentifier, or something. Those constants are 100% related to NostrIdentifier and nothing else.

### CODE REVIEW

File: Nos/Models/NoteParser.swift [hunk 1]
```diff
@@ -137,6 +137,8 @@ struct NoteParser {
                 case .naddr(let replaceableID, _, let authorID, let kind):
                     return "\(prefix)[\(String(localized: .localizable.linkToNote))]" +
                         "($\(replaceableID);\(authorID);\(kind))"
+                case .nsec:
+                    return String(substring)
```
Comment (mplorentz) [inline Nos/Models/NoteParser.swift:141]: haha, I hope we never hit this case 🤦‍♂️

**Review by mplorentz — APPROVED** [review]
This is 💯 

But I found a bug I didn't notice when reviewing your last PR but I imagine it must have come from there. When I tap the links in [this note](https://njump.me/note1t3akhmrnytty3lh73kuxslj2ez7kctvkt5ju6kyvnll055qjfhgsu9cgs4) in Nos Dev they open in Nos Staging. Have you seen that? Can we fix that as part of one of the naddr tickets?

**Comment by joshuatbrown** [comment]
> This is 💯
> 
> But I found a bug I didn't notice when reviewing your last PR but I imagine it must have come from there. When I tap the links in [this note](https://njump.me/note1t3akhmrnytty3lh73kuxslj2ez7kctvkt5ju6kyvnll055qjfhgsu9cgs4) in Nos Dev they open in Nos Staging. Have you seen that? Can we fix that as part of one of the naddr tickets?

I think we're out of `naddr` tickets at this point, other than this one, so I'll do it in here.

**Comment by joshuatbrown** [comment]
@mplorentz I'd love another review when you can. I fixed the issue you pointed out and renamed `Nostr` to `NostrIdentifierPrefix`.

**Review by mplorentz — APPROVED** [review]
🚀

**Pull request merged**